specifyweb/stored_queries/queryfieldspec.py

```diff
@@ -177,33 +181,33 @@ def from_stringid(cls, stringid, is_relation):
         extracted_fieldname, date_part = extract_date_part(field_name)
         field = node.get_field(extracted_fieldname, strict=False)
 
```

Comment:
Yikes, you're quite brave for merging production in this branch. Most of the code looks not too bad. Except this one, where I thought some insider info will be necessary.

I know getting rid of TreeRankQuery seems like an easy way, but trust me, TreeRankQuery makes batch-edit very simple. Don't take my word for it. Go to the sibling file batch_edit.py, and you'll see just how minimal changes were necessary to support dataset construction when trees are selected. (IIRC, there are like just 3 places, quite isolated from an abstraction perspective).

Here's the idea behind TreeRankQuery. Basically, each rank is considered as a relationship from tree to itself. So, Kingdom is a to-one relationship from Taxon to Taxon.  So, when user selects Collectionobject -> Determination -> Taxon (species, fullname), join path becomes

```
determination, taxon, species, fullname
```

Pros
1. No need for tree_rank and tree_field fields.
2. It's quite easy to test if join path ends with relationship (just check the last field is relationship or not). You'd previously also have to think about tree_rank and tree_field.
3. The code, before this merge, **already** constructs correct queries for something like `determination, taxon, species, createdby, firstname`, effectively allowing relationships from tree ranks. Try doing that with tree_rank and tree_field!
4. Most importantly (/s), it makes batch-edit quite simple. For every row, batch-edit dataset construction needs to look at what the IDs are. When you have TreeRankQuery, tree ranks are effectively just like any other relationship. 

Cons:
1. Hard to merge from production, which is a valid reason

Reply:
Thanks! I was reluctant to merge prod here initially but the alternative seemed to be a bigger headache. I'll look into incorporating TreeRankQuery in the PR that follows #5417